src.ts/wallet/utxohdwallet.ts

```diff
@@ -108,34 +117,21 @@ export class UTXOHDWallet extends BaseWallet {
 
      coinType?: number;
 
-     /** 
-      * array of addresses/priv keys this current wallet is aware of that were 
-      * generated by incrementing addressindex in bip44 until specified gap rule
-      */
-    #utxoAddresses: UTXOAddress[] = [];
-
-    get utxoAddresses(): UTXOAddress[] {
-        return this.#utxoAddresses;
-    }
-
-    set utxoAddresses(addresses: UTXOAddress[]) {
-        this.#utxoAddresses = addresses;
-    }
-
-    // map of addresses to unspent outputs
-    #addressOutpoints: { [address: string]: Outpoint[] } = {};
+    /**
+     * Map of shard name (zone) to shardWalletData
+     * shardWalletData contains the addresses and outpoints for the shard
+     * that are known to the wallet
+     */
+    #shardWalletsMap: Map<string, ShardWalletData> = new Map();
 
-    get addressOutpoints(): { [address: string]: Outpoint[] } {
-        return this.#addressOutpoints;
+    get shardWalletsMap(): Map<string, ShardWalletData> {
```

Comment:
I'm not sure we want to provide a public getter for private key data. Is there a reason a user would want to access the data in `shardWalletData` directly?

Reply:
You are right we can revisit.